HHS/Head-Start-TTADP
# Add loader, bug fixes, payload reduction

## Description of change

Add a loading overlay to be shown when components are loading. Added the loader to the `Container` component so it dims the complete container when loading. Widgets now have default data defined so they can be displayed with "null" data. This should stop the UI from "jumping" around so much as data is loaded in and components are resized.

Removed some old code that is no longer needed (data scripts that have already been ran against prod).

- This change will also make the colors on the accessible table closer match the colors on the activity report table, and the aria-labels will more accurately report the function of the individual apply filters buttons to screenreaders.
-  When updating an activity report, only the updated fields are sent to the backend and not the entire report data, in order to  reduce data submission size. 
- A bug where a less than thirty-one day range that stretched across two months would be incorrectly labelled in the accessible data table for the total hours and grantee graph has been fixed. 
- An instruction label ("Select the date range for TTA activities") has been added to the date range select. This language will be updated in the next PR.

## How to test

Browse the regional dashboard on [staging](https://tta-smarthub-staging.app.cloud.gov/regional-dashboard). Note when you change filters or load the page a faint white overlay is shown and elements inside are not click-able. The tables/widget on the activity reports page also has the white overlay when new data is being loaded.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-152
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-222
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-323
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-222

## Checklists

### Every PR


- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy


### After merge/deploy
